galaxy_ng/app/auth/ldap.py

```diff
@@ -40,3 +40,22 @@ class GalaxyLDAPBackend(LDAPBackend):
 
     def __init__(self):
         self.settings = GalaxyLDAPSettings(self.settings_prefix, self.default_settings)
+
+
+class PrefixedLDAPBackend(GalaxyLDAPBackend):
+
+    @property
+    def prefix(self):
+        return settings.get("RENAMED_USERNAME_PREFIX")
+
+    def authenticate(self, *args, **kwargs):
+        if username := kwargs.get("username"):
+            if username.startswith(self.prefix):
```

Comment:
Wouldn't this refuse to authenticate a user if their username didn't start with the prefix?

What if a user is not a conflict. It only ever existed in Hub, and then it is migrated. That user won't have the prefix. The way this is changing looks like it might block such a user from being able to log in with LDAP. Because this replaces the old `GalaxyLDAPBackend`, instead of adding to it.

I'm also somewhat expecting to save a variable on `self` to indicate whether or not the prefix was removed in this step, and then that's what we would use to determine whether or not the prefix is added back in `get_or_build_user`.

Reply:
The first problem has been fixed. The second comment is more complex...

Setting self.prefix_removed does the right thing when authenticating with prefix_username:password, but won't do the right thing when authenticating with username:password ... it ends up creating a second user with a new uid.